pkg/generators/servers.go

```diff
@@ -330,19 +331,59 @@ func (g *ServersGenerator) generateServerAdapterSource(resource *concepts.Resour
 			{{ $requestBodyParameters := requestBodyParameters . }}
 			{{ $requestBodyLen := len $requestBodyParameters }}
 			{{ $responseParameters := responseParameters . }}
+			{{ $requestQueryParameters := requestQueryParameters . }}
 	
 			func (a *{{ $adapterName }}) read{{ $requestName }}(r *http.Request) (*{{ $requestName }}, error) {
+				var err error
 				result := new({{ $requestName }})
-				result.query = r.Form
+				result.query = r.URL.Query()
 				result.path = r.URL.Path
 
+				{{ if $requestQueryParameters }}
+					{{ range  $requestQueryParameters }}
+						{{ $fieldName := fieldName . }}
+						{{ $fieldType := fieldType . }}
+						{{ $fieldName }}String := result.query.Get("{{ $fieldName }}")
+						if {{ $fieldName }}String  != "" {
+							{{ if eq .Type.Name.String "integer"}}
+								{{ $fieldName }}, err := strconv.Atoi({{ $fieldName }}String)
+								if err != nil {
+									return nil, err
+								}
+								result.{{ $fieldName }} = &{{ $fieldName }}
+
+							{{ end }}
+							{{ if eq .Type.Name.String "float" }}
+								{{ $fieldName }}, err = strconv.ParseFloat({{ $fieldName }}String, 64)}
```

Comment:
I think this and cases below should be `:=`.
If not, then L349 above should be `=`.